test.py

```diff
@@ -12,5 +12,5 @@
 
 model = StackedAutoEncoder(dims=[200, 200], activations=['linear', 'linear'], epoch=[
                            3000, 3000], loss='rmse', lr=0.007, batch_size=100, print_step=200)
-model.fit(train_X)
```

Comment:
Maybe rename test_X to valid_X ? Can you please keep both the usages (w/ and w/o valid_x) ?

Reply:
Yes that's a good idea. 
Maybe also add an example for fit_transform()?